test-lib.sh

```diff
@@ -456,13 +456,34 @@ ct_s2i_build_as_df()
     local user_id=
     local df_name=
     local tmpdir=
+    local incremental=false
     # Use /tmp to not pollute cwd
     tmpdir=$(mktemp -d)
     df_name=$(mktemp -p "$tmpdir" Dockerfile.XXXX)
     pushd "$tmpdir"
     # Check if the image is available locally and try to pull it if it is not
     docker images "$src_image" &>/dev/null || echo "$s2i_args" | grep -q "pull-policy=never" || docker pull "$src_image"
-    user_id=$(docker inspect -f "{{.Config.User}}" "$src_image")
+    user=$(docker inspect -f "{{.Config.User}}" "$src_image")
+    # Default to root if no user is set by the image
+    user=${user:-0}
+    # run the user through the image in case it is non-numeric or does not exist
+    if ! [ "$user" -eq "$user" ] 2>/dev/null || ! user_id=$(docker run --rm "$src_image" bash -c "id -u $user 2>/dev/null"); then
```

Comment:
`s/||/&&/` ?  btw. interesting hack with `-eq` :-)

Reply:
Ah, yes it should be `&&`. Also missing an `else` for when the condition fails on the `-eq` check to get the already numeric value into `$user_id`. Will also add a note explaining the use of `-eq` in the test.